lib/puppet/type/postgresql_psql.rb

```diff
@@ -48,6 +48,28 @@ def matches(value)
     end
   end
 
+  newparam(:onlyif) do
+    desc "An optional SQL command to execute prior to the main :command; " +
+        "this is generally intended to be used for idempotency, to check " +
+        "for the existence of an object in the database to determine whether " +
+        "or not the main SQL command needs to be executed at all."
+
+    # Return true if a matching row is found
+    def matches(value)
+      if Puppet::PUPPETVERSION.to_f < 4
```

Comment:
Why are you checking against puppet 4? run_unless_sql_command will return an array for version greater than 3.4